<h1> Lab 6 </h1> 
<h3> Akshita Gundavarau

<h2> 1 Confusion Matrix </h2>
<h3> 1.1


<h5> M1: </h5>

| | **Predicted** | |
| -------- | ------------: | ------:|
| | Poisonous | Edible |
|**Actual**| 3 | 0 |
||3|4|

<h5> M2: </h5>

| | **Predicted** | |
| -------- | ------------: | ------:|
| | Poisonous | Edible |
|**Actual**| 2 | 1 |
||0|7|

<h3> 1.2

Accuracy = (True poisonous + True edible) / Total 

Precision = (True Edible / Total Predicted Edible)

Recall = (True Edible / Total Actual Edible)

<h4> M1: </h4>

In [46]:
M1_accuracy = (4+3)/10
M1_accuracy

0.7

In [47]:
M1_precision = 4/4
M1_precision

1.0

In [48]:
M1_recall = 4/7
M1_recall

0.5714285714285714

<h4> M2: </h4>

In [49]:
M2_accuracy = (7+2)/10
M2_accuracy

0.9

In [50]:
M2_precision = 7/8
M2_precision

0.875

In [51]:
M2_recall = 7/7
M2_recall

1.0

<h3> 1.3 </h3> 

Which model, M1 or M2 will you recommend him to use? Explain your reasoning!

For this question. Having a false positive is bad. This would mean that the model predicts that the mushroom is edible but it is actually poisonous. Precision measures the percentage of predicted positives that are captured. 

M1 had a precision of 100% while M2 has a precision of 87.5%. 

Therefore, I would suggest to use Model 1. 


<h3> 1.4 </h3> 

<h5> M3: </h5>

| | **Predicted** | |
| -------- | ------------: | ------:|
| | Innocent | Guilty |
|**Actual**| 3 | 2 |
||1|4|

<h5> M4: </h5>

| | **Predicted** | |
| -------- | ------------: | ------:|
| | Innocent | Guilty |
|**Actual**| 5 | 0 |
||3|2|

<h3> 1.5 </h3>

Accuracy = (True Guilty + True Innocent) / Total 

Precision = (True Guilty / Total Predicted Guilty)

Recall = (True Guilty / Total Actual Guilty)

<h4> M3: </h4>

In [52]:
M3_accuracy = (4+3)/10
M3_accuracy

0.7

In [1]:
M3_precision = 4/6
M3_precision

0.6666666666666666

In [2]:
M3_recall = 4/5
M3_recall

0.8

<h4> M4: </h4> 

In [55]:
M4_accuracy = (2+5)/10
M4_accuracy

0.7

In [3]:
M4_precision = 2/2
M4_precision

1.0

In [4]:
M2_recall = 2/5
M2_recall

0.4

<h3> 1.6 </h3> 

Which model would you recommend her to use? Explain your reasoning!

For this question. Having a false positive is bad. This would mean that the model predicts that the someone is guilty but they are actually innocent. Precision measures the percentage of predicted positives that are captured. 

M4 had a precision of 100% while M3 has a precision of 66.6%. 

Therefore, I would suggest to use Model 4. 


<h3> 1.7 </h3> 

<h5> M5: </h5>

| | **Predicted** | |
| -------- | ------------: | ------:|
| | Others | Simolestes Vorax  |
|**Actual**| 3 | 2 |
||1|4|

<h5> M6: </h5>

| | **Predicted** | |
| -------- | ------------: | ------:|
| | Others | Simolestes Vorax  |
|**Actual**| 2| 3 |
||0|5|

<h3> 1.8 </h3>

Accuracy = (True Simolestes Vorax  + True Others) / Total 

Precision = (True Simolestes Vorax / Total Predicted Simolestes Vorax)

Recall = (True Simolestes Vorax / Total Actual Simolestes Vorax)

F-score = 2 / ( (1 / precision) + (1 / recall) )

<h4> M5: </h4>

In [58]:
M5_accuracy = (3+4)/10
M5_accuracy

0.7

In [59]:
M5_precision = 4/6
M5_precision

0.6666666666666666

In [60]:
M5_recall = 4/5
M5_recall

0.8

In [61]:
M5_f_score = 2 / ((1/M5_precision) + (1/M5_recall))
M5_f_score

0.7272727272727273

<h4> M6: </h4>

In [62]:
M6_accuracy = (2+5)/10
M6_accuracy

0.7

In [63]:
M6_precision = 5/8
M6_precision

0.625

In [64]:
M6_recall = 5/5
M6_recall

1.0

In [65]:
M6_f_score = 2 / ((1/M6_precision) + (1/M6_recall))
M6_f_score

0.7692307692307692

<h3> 1.8 </h3> 

Which of these two models, M5 or M6 will you recommend him? Explain to him.

For this question it is hard to determine if having a false positive is worse than having a false negative. Hence we will use the F-score values (harmonic mean of Precision, Recall) to figure out which model is better. Overall, it is better to have a higher precision and higher recall. The F-score will  

The F-score of M5 is 72.7%. The F-score of MS is 